src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -281,7 +281,7 @@ public bool TryGetSemanticModel([NotNullWhen(returnValue: true)] out SemanticMod
                 var syntaxTree = await this.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
                 var compilation = (await this.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false))!;
 
-                var result = compilation.GetSemanticModel(syntaxTree);
+                var result = compilation.GetSemanticModel(syntaxTree!);
```

Comment:
@jasonmalinowski , @sharwell wanted to make sure you all saw this. It seemed suspicious at a glance.

Reply:
@jaredpar: you can change the call on line 281 to GetRequiredSyntaxTreeAsync(), which we've since added as an internal helper to avoid this problem.